stdlib/public/Concurrency/AsyncStream.swift

```diff
@@ -12,95 +12,168 @@
+///     }
+///
+/// To adapt this to use `async`-`await`, extend the `QuakeMonitor` to add a
+/// `quakes` property, of type `AsyncStream<Quake>`. In the getter for this
+/// property, return an `AsyncStream`, whose `build` closure -- called at
+/// runtime to create the stream -- does the following:
+///
+/// 1. Creates a `QuakeMonitor` instance.
+/// 2. Sets the monitor's `quakeHandler` property to a closure that receives
+/// each `Quake` instance and forwards it to the stream by calling the
+/// continuation's `yield(_:)` method.
+/// 3. Sets the continuation's `onTermination` property to a closure that
+/// calls `stopMonitoring()` on the monitor.
+/// 4. Finally, calls `startMonitoring` on the `QuakeMonitor`.
+///
+///     extension QuakeMonitor {
+///
+///         static var quakes: AsyncStream<Quake> {
+///             AsyncStream { continuation in
+///                 let monitor = QuakeMonitor()
+///                 monitor.quakeHandler = { quake in
+///                     continuation.yield(quake)
+///                 }
+///                 continuation.onTermination = { @Sendable _ in
+///                     monitor.stopMonitoring()
+///                 }
+///                 monitor.startMonitoring()
+///             }
 ///         }
-///         continuation.finish()
-///       }
 ///     }
 ///
-///     for await digit in digits {
-///       print(digit)
+/// Since the stream is an `AsyncSequence`, the call point can use the
+/// `for`-`await`-`in` syntax to process each `Quake` instance as the stream
+/// produces it:
+///
+///     for await quake in QuakeMonitor.quakes {
+///         print ("Quake: \(quake.date)")
 ///     }
+///     print ("Stream finished.")
 ///
 @available(SwiftStdlib 5.5, *)
 public struct AsyncStream<Element> {
+    
+  /// A mechanism to interface between synchronous code and an asynchronous
+  /// stream.
+  ///
+  /// The closure you provide to the `AsyncStream` in
+  /// `init(_:bufferingPolicy:_:)` receives an instance of this type when
+  /// invoked. Use this continuation to provide elements to the stream by
+  /// calling one of the `yield` methods, then terminate the stream normally by
+  /// calling the `finish()` method.
   public struct Continuation: Sendable {
-    /// Indication of the type of termination informed to
-    /// `onTermination`.
+    /// A type that indicates how the stream terminated.
+    ///
+    /// The `onTermination` closure receives an instance of this type.
     public enum Termination {
       
-      /// The stream was finished via the `finish` method
+      /// The stream finished as a result of calling the continuation's
+      ///  `finish` method.
       case finished
       
-      /// The stream was cancelled
+      /// The stream finished as a result of cancellation.
       case cancelled
     }
     
-    /// A result of yielding values.
+    /// A type that indicates the result of yielding a value to a client, by
+    /// way of the continuation.
```

Comment:
Added a discussion below. There are three flavors of `yield` in `AsyncStream` (and two in `AsyncThrowingStream`), so I just used code-voice rather than linking to all of them.